src/lib/api/CollectionApi.ts

```diff
@@ -0,0 +1,26 @@
+import appConfig from "~/config/appConfig";
+import { CollectionQuery, CollectionResult } from "~/src/types/CollectionQuery";
+
+const apiEnv = process.env["APP_ENV"];
+const apiUrl = process.env["API_URL"] || appConfig.api.url[apiEnv];
+
+const { collectionPath } = appConfig.api;
+const collectionUrl = apiUrl + collectionPath;
+export const collectionFetcher = async (query: CollectionQuery) => {
+  const collectionApiQuery = {
+    ...query,
+  };
+
+  const url = new URL(collectionUrl + "/" + query.identifier);
+  url.search = new URLSearchParams(collectionApiQuery.toString()).toString();
+  const res = await fetch(url.toString());
+
+  if (res.ok) {
+    const collectionResult: CollectionResult = await res.json();
```

Comment:
nit: it could be a good idea to wrap this `await res.json()` in a try/catch block just in case something goes awry with the json parsing